src/components/Task/Modal/TaskModal.tsx

```diff
@@ -45,69 +52,63 @@ const TaskModal = ({
 
   const task = data?.task;
 
-  const renderTitle = (): string => {
+  const renderTitle = useMemo((): string => {
```

Comment:
If you're keeping the memo, can you rename this variable since it's a string now and not a function?

```suggestion
  const title = useMemo((): string => {
```